### DISCUSSION TITLE
How come can I ping both meshes with TREL (with diagram)

### DISCUSSION THREAD

**Discussion by fgervais** [openthread/openthread#8692] [Q&A] [answered]
I created a basic test network with 2 border routers physically far appart with TREL enabled.

I did not set any prefix before `thread start` and so I assume they somehow agreed on a common one over TREL.

The resulting topology is the following:

![image](https://user-images.githubusercontent.com/5411996/214688161-70e7350a-48b1-44b3-9ee0-ecfbd6dbdc3c.png)

Since they both have the same prefix, I assume they both advertise as handling this network and so when I look in the `PC` routing table I see the "winner". As seen in the graph, it's `OTBR1`.

From here I thought that the `PC` would only be able to ping `fddd:381d:e1ec:1:607d:d12c:ab15:b069` under `OTBR1` but it is somehow not the case.

How is it possible that the `PC` can also ping `fddd:381d:e1ec:1:4303:c58d:ee48:d3fd` under `OTBR2`.

**Comment by jwhui** [comment, marked as answer]
The purpose of TREL is to incorporate adjacent infrastructure links (e.g. Wi-Fi or Ethernet) into the Thread topology. With TREL, OTBR1 and OTBR2 are able to discover and form links using standard Thread protocols (i.e. MLE). In this case, packets sent from PC to `fddd:381d:e1ec:1:4303:c58d:ee48:d3fd` will route through OTBR1 and OTBR2.

**Reply by fgervais** [reply]
Thank you very much for clarifying.

**Reply by dounie** [reply]
Thanks for this clarifying answer. Could you confirm I got that right : MLE packets should be sent over TREL (UDP/IPv6/Ethernet) to ensure the "discovery" of OTBR inbetween them ? 
Currently, I have to feed them with the same dataset so that they discover themselves, so I guess I'm doing something wrong !

**Reply by fgervais** [reply]
I think it's fine to have to feed the dataset as it is how they will know they fit together.

I think TREL is really to be seen as the same as the mesh but over an alternate path. Once they can see them in the mesh, they can see them over TREL.

**Reply by jwhui** [reply]
Yes, TREL makes it possible to incorporate Wi-Fi/Ethernet links into the Thread mesh topology. TREL basically transports 15.4 frames in UDP packets. As a result, TREL also transports MLE messages. Devices participating in the same Thread network should be configured with the same Active Operational Dataset.
